packages/eas-cli/src/commandUtils/flags.ts

```diff
@@ -10,6 +20,53 @@ export const EasNonInteractiveAndJsonFlags = {
   }),
 };
 
+export const EASEnvironmentFlag = {
+  environment: Flags.enum({
+    description: "Environment variable's environment",
+    parse: upperCaseAsync,
+    options: mapToLowercase([
+      EnvironmentVariableEnvironment.Development,
+      EnvironmentVariableEnvironment.Preview,
+      EnvironmentVariableEnvironment.Production,
+    ]),
+  }),
+};
+
+export const EASVariableFormatFlag = {
+  format: Flags.enum({
+    description: 'Output format',
+    options: ['long', 'short'],
+    default: 'short',
+  }),
+};
+
+export const EASVariableVisibilityFlags = {
+  sensitive: Flags.boolean({
+    description: 'Encrypt variable value at rest',
+    default: false,
+    exclusive: ['secret', 'public'],
+  }),
+  secret: Flags.boolean({
+    description: 'Encrypt variable value at rest and protect it from being displayed',
+    default: false,
+    exclusive: ['sensitive', 'public'],
+  }),
+  public: Flags.boolean({
+    description: 'Make variable value public',
+    default: false,
+    exclusive: ['sensitive', 'secret'],
+  }),
+};
```

Comment:
While I agree that your explanation makes sense and when using --sensitive` you don't need to remember about the visibility concept, it's not a standard practice across the EAS CLI codebase.

I mean we don't do `eas build --android`, but `eas build --platform android`.

In my opinion, it would be better to do `--visibility sensitive|secret|public`, but if you strongly disagree, that's not a hill I'm going to die on.

CC: @radoslawkrzemien what do you think?

Reply:
Using these 3 flags instead of 1 `visibility` flag makes it harder to describe what the default is. It's not clear looking at these 3 flags that the default behavior is `public`, while if you do the `visibility` flag with `default: 'public'` it's entirely clear.

Reply:
We also have `--scope shared|project` vs `--shared` and `--project` flags, it seems a little bit inconsistent